src/components/SuperAdmins/index.js

```diff
@@ -1,101 +1,129 @@
+    openModal();
+  };
+
+  const search = (value) => {
+    setSearchQuery(value);
+    setFilteredList(
+      superAdmins.filter((item) => item.firstName.toLowerCase().includes(value.toLowerCase()))
     );
   };
 
-  const openModal = (id) => {
-    setToBeDeleted(id);
+  const openModal = () => {
     setIsOpen(true);
   };
 
-  useEffect(async () => {
-    try {
-      setLoading(true);
-      const response = await fetch(`${process.env.REACT_APP_API_URL}/super-admins`);
-      const data = await response.json();
-      setDataTable(data.data);
-      setUntouchedData(data.data);
-      setLoading(false);
-    } catch (error) {
-      console.error(error);
-    }
-  }, []);
-
-  const deleteSuperAdmin = async (id) => {
-    setLoading(true);
-    const response = await fetch(`${process.env.REACT_APP_API_URL}/super-admins/${id}`, {
-      method: 'DELETE'
-    });
-    const res = await response.json();
-    if (res.data.error) {
-      setIsOpen(false);
-      setLoading(false);
-    } else {
-      setDataTable([...dataTable.filter((admin) => admin._id !== id)]);
-      setIsOpen(false);
-      setLoading(false);
-    }
+  const closeModal = () => {
+    setIsOpen(false);
   };
 
-  const column = [
-    { heading: 'Id', value: '_id' },
-    { heading: 'First Name', value: 'firstName' },
-    { heading: 'Last Name', value: 'lastName' },
-    { heading: 'Email', value: 'email' }
-  ];
   if (loading) {
     return (
       <div className={styles.loadingDiv}>
-        <LoadingScreen />;
+        <LoadingScreen />
       </div>
     );
   }
   return (
     <section className={styles.containerSuperAdmin}>
       <Modal
+        modalTitle={'Super Admins'}
         isOpen={isOpen}
-        handleClose={() => setIsOpen(!isOpen)}
-        modalTitle={'Delete Super Admin'}
+        handleClose={() => {
+          openModal();
+        }}
       >
-        <p>Do you Wish to delete the Super Admin?</p>
+        <p>{error ? error : 'Are you sure to delete a Super Admin?'}</p>
         <div>
-          <Button type={'delete'} handler={() => deleteSuperAdmin(toBeDeleted)} text={'Yes'} />
-          <Button handler={() => setIsOpen(!isOpen)} text={'No'} />
+          {error ? (
+            <div>
+              <Button text="Close" handler={closeModal} />
+            </div>
+          ) : (
+            <div>
+              <Button
+                text="Yes"
+                type="delete"
+                handler={() => {
+                  dispatch(deleteSuperAdmin(superAdminId));
+                  closeModal();
+                }}
+              />
+              <Button text="No" handler={closeModal} />
+            </div>
+          )}
         </div>
       </Modal>
       <h2>Super Admins</h2>
       <div className={styles.btnSearchDiv}>
-        <Button link={'/super-admins/form'} text={'Add Super Admin'} />
+        <Button
+          text={'Add Super Admin'}
+          handler={() => {
+            dispatch(cleanSuperAdmin());
```

Comment:
Could you check if you can do this?
```suggestion
            dispatch(setSuperAdmin());
```
This should work since we have a ternary on the reducer that if we don't send anything we set the superAdmin an object empty

And if it works, we can remove the `cleanSuperAdmin` since both do the same.